### ISSUE TITLE
List assistants throwing error when empty list.

### ISSUE THREAD

**Issue by akshdeep-singh** [davidmigloz/langchain_dart#604]
### System Info

openai_dart: 0.4.4

### Related Components

- [ ] doc-loaders
- [ ] doc-transformers
- [ ] prompts
- [ ] llms
- [ ] chat-models
- [X] output-parsers
- [ ] chains
- [ ] memory
- [ ] stores
- [ ] embeddings
- [ ] retrievers
- [ ] agents

### Reproduction

1. Use the [List assistants](https://platform.openai.com/docs/api-reference/assistants/listAssistants) api without creating any assistant.

Following code in the package throws error:
```dart
_$ListAssistantsResponseImpl _$$ListAssistantsResponseImplFromJson(
        Map<String, dynamic> json) =>
    _$ListAssistantsResponseImpl(
      object: json['object'] as String,
      data: (json['data'] as List<dynamic>)
          .map((e) => AssistantObject.fromJson(e as Map<String, dynamic>))
          .toList(),
      firstId: json['first_id'] as String,
      lastId: json['last_id'] as String,
      hasMore: json['has_more'] as bool,
    );
```

Error reason: When list is empty, the `first_id` and `last_id` are null:

![image](https://github.com/user-attachments/assets/f0569811-228a-4948-bf78-bde98e98560d)

Solution: Change `String` to `String?`.

### Expected behavior

Should not throw error when parsing json response.

**Comment by davidmigloz** [comment]
hey @akshdeep-singh ,

Thanks for reporting, I'll make it nullable in the next release.

**Comment by davidmigloz** [comment]
Fixed in [openai_dart 0.4.5](https://pub.dev/packages/openai_dart/changelog#045)
